stripped/colabs/Regresja_logistyczna_i_SVM_stripped.py

```diff
@@ -0,0 +1,1180 @@
+
+OLS_report_and_ANOVA(winter_data, 'pustynia', 'PotEvap')
+
+perform_Kruskal_Wallis(winter_data, 'pustynia', 'PotEvap', 13)
+
+test_for_residuals_normality(winter_data, 'RootMoist')
+
+plot_density(winter_data, 'pustynia', 'RootMoist', random_state=13)
+
+OLS_report_and_ANOVA(winter_data, 'pustynia', 'RootMoist')
+
+perform_Kruskal_Wallis(winter_data, 'pustynia', 'RootMoist', 13)
+
+test_for_residuals_normality(winter_data, 'SoilM_100_200cm')
+
+plot_density(winter_data, 'pustynia', 'SoilM_100_200cm', random_state=13)
+
+OLS_report_and_ANOVA(winter_data, 'pustynia', 'SoilM_100_200cm')
+
+perform_Kruskal_Wallis(winter_data, 'pustynia', 'SoilM_100_200cm', 13)
+
+full_summer_data = spark.sql('''SELECT * from nasa_full WHERE Month = 7 AND (Year = 2021 OR Year = 2019)''').toPandas()
+full_summer_data = full_summer_data.drop(columns=['Month', 'Year', 'lon', 'lat'])
+
+corr = full_summer_data.corr(method = "pearson")
+cmap = sns.diverging_palette(250, 323, 80, 60, center='dark', as_cmap=True)
+sns.heatmap(corr, vmax =1, vmin=-0.3, cmap = cmap, square = True, linewidths = 0.2, annot=True)
+
+sns.pairplot(data=full_summer_data, diag_kind='kde')
+
+full_winter_data = spark.sql('''SELECT * from nasa_full WHERE Month = 12 AND (Year = 2021 OR Year = 2019)''').toPandas()
+full_winter_data = full_winter_data.drop(columns=['Month', 'Year', 'lon', 'lat'])
+
+corr = full_winter_data.corr(method = "pearson")
+cmap = sns.diverging_palette(250, 323, 80, 60, center='dark', as_cmap=True)
+sns.heatmap(corr, vmax =1, vmin=-0.3, cmap = cmap, square = True, linewidths = 0.2, annot=True)
+
+sns.pairplot(data=full_winter_data, diag_kind='kde')
+
+def Kfolds_crossvalidation( model: BaseEstimator, folds: np.ndarray, folds_labels: List[List[int]]) -> dict:
+
+ validation_accuracy = []
+ validation_recall = []
+ validation_precision = []
+ validation_ROCAUC = []
+ validation_jaccard = []
+
+ k=len(folds)
+ for i in range(k):
+       X_test = folds[i]
+       y_test = folds_labels[i]
+       train_folds = [fold for fold in folds if fold.tolist() != folds[i].tolist()]
+       train_labels = [labels for labels in folds_labels if labels != folds_labels[i]]
+       X_train = np.concatenate(train_folds)
+       y_train = np.concatenate(train_labels)
+       trained_model = model.fit(X_train, y_train)
+       y_pred = trained_model.predict(X_test)
+       validation_accuracy.append(accuracy_score(y_test, y_pred))
+       validation_recall.append(recall_score(y_test, y_pred))
+       validation_precision.append(precision_score(y_test, y_pred, zero_division=0))
+       validation_ROCAUC.append(roc_auc_score(y_test, y_pred))
+       validation_jaccard.append(jaccard_score(y_test, y_pred))
+
+       results = {'accuracy': mean(validation_accuracy),
+                 'recall': mean(validation_recall),
+                 'precision': mean(validation_precision),
+                  'ROC-AUC': mean(validation_ROCAUC),
+                  'jaccard_score': mean(validation_jaccard)}
```

Comment:
Czy to wcięcie jest poprawne?

Reply:
Wcięcia to bloki, a ten był w pętli. W pętli było tworzone wiele różnych obiektów result.

Reply:
A, już rozumiem pytanie, myślałam, ze to był inny fragment kodu. Tak, racja, jakoś mi się przypadkiem to musiało przesunąć, nie powinno być w pętli - ale szczęśliwie nie wplynęło to na końcowy wynik (po prostu zmienna się aktualizuje w każdej iteracji, ale koniec końców wynik będzie ten sam). Poprawię.